### ISSUE TITLE
Prefixed CSS properties' tables are unclear

### ISSUE THREAD

**Issue by iherman** [w3c/epub-specs#1390]
I am not sure I get the intention of the tables in [§5.3.3. Prefixed properties](https://w3c.github.io/epub-specs/epub33/core/#sec-css-prefixed). I _suspect_ the message is:

- in earlier versions of EPUB 3 prefixed CSS properties were used because there were no final CSS properties for the specific functionalities
- these functionalities now exists in CSS 3
- new contents should use the standard functionalities, but the old, prefixed properties remain valid (and the RS conformance says that RS-s must continue to accept and implement those)
- here are the tables how to map the old properties, and their allowed values, to the final CSS properties and their values

If that is the case, I do not really understand 

- what "Prior EPUB Mapping" really means
- what does "no change" means under the heading "Mapping to CSS Text 3"
- it is also not really clear from the table what the corresponding property is (although, in most of the cases, this is probably the one that one gets dropping the `-epub-` prefix, but not always)

I would look for a table somewhat of the sort:

| Prefixed property | Value | Standard CSS property | Value |
| :--------------- | :----| :------------------- | :--- |
| `-epub-text-orientaton` | `upright` | `text-orientation` | `upright` |
| etc |
| `-epub-text-combine-horizontal` | `none` | `text-combine-upright` | `none` |
| etc |
| `-epub-hyphens` | `none` \| `manual` \| `auto` | `hyphens` | `none` \| `manual` \| `auto`
| etc |

I realize the table includes some items that are considered to be deprecated; imho these should be part of a separate table.

Does this make sense?

**Comment by dauwhe** [comment]
This is all my fault. I spent an immense amount of time trying to trace the history of these properties and values. The CSS Writing modes spec has changed A LOT over the years. The epub-prefixed properties were taken from an earlier version of writing modes, but by the time 3.1/3.2 came along there had been significant changes. So it seemed important to document this, and provide some guidance to content authors who should move away from the prefixed properties. 

I coordinated quite a bit with Murata-san and others who knew what properties were actually supported by Japanese reading systems. The deprecated values here were not supported, and were not in the EBPAJ documents, and so we could deprecate them without affecting any existing content. 

So, for example, one possible value of `-epub-text-orientation` is `mixed`. EPUB continues to support this value, even though the best practice would be to use the unprefixed `text-orientation: mixed`. But there was also an older value allowed in EPUB: `-epub-text-orientation: vertical-right`. This has been deprecated. 

Basically the table says, here are all the prefixed properties and values. Find the values you are using in your ebook. If it's one of the deprecated ones in the "prior epub mapping" column, here's what to change it to. If you want to move away from the prefixed properties, this will also tell you which unprefixed property to use.

**Comment by frivoal** [comment]
I wonder if it would be a good idea, after doing some analysis of existing content and existing readers to identify / confirm which of these properties and values are necessary for compat, to upstream their definitions into the css specs themselves. We're already doing that for the -webkit- properties that cannot be removed, so maybe we should consider whether a similar logic applies here. Or is this specialized enough that the listing and the mapping should stay where it is?

**Comment by iherman** [comment]
@dauwhe I certainly appreciate the difficulties with all this and that it is also necessary to have this, but what I can say as a freshman reading through the document in details, it is very difficult to understand what all this is for. 

The reference to an outdated specification is also troubling. The message it conveys is that (1) you should use the outdated spec because this is what the EPUB spec refers to and then (2) RS-s have to implement an outdated spec of CSS. Obviously, (2) is actually quite impossible because if a RS is built on the latest version of, say, chromium, then this is not the version that would get implemented.

I am not sure how, but I believe this section may need further re-write... For example:

> So, for example, one possible value of `-epub-text-orientation` is `mixed`. EPUB continues to support this value, even though the best practice would be to use the unprefixed `text-orientation: mixed`. But there was also an older value allowed in EPUB: `-epub-text-orientation: vertical-right`. This has been deprecated. 
> 
> Basically the table says, here are all the prefixed properties and values. Find the values you are using in your ebook. If it's one of the deprecated ones in the "prior epub mapping" column, here's what to change it to. If you want to move away from the prefixed properties, this will also tell you which unprefixed property to use. 

makes it indeed a bit clearer, but such explanation, at the minimum, should be in the spec text...

**Comment by lilles** [comment]
I found this issue when someone asked what these -epub- properties are and whether we ship them. Chromium currently ships -epub-* properties in the ref below, all as aliases for something else:

https://source.chromium.org/chromium/chromium/src/+/487f2aa8:third_party/blink/renderer/core/css/css_properties.json5;l=6382-6418

**Comment by murata2makoto** [comment]
@frivoal Although it is difficult to examine all EPUB publications in Japan, we know the CSS stylesheets defined by the EBPAJ EPUB profile.   We have assumed that all CSS properties occurring in these CSS stylesheets are indeed used.   The current table is based on this assumption.

**Comment by dauwhe** [comment]
Closed via #1655
